### DISCUSSION TITLE
Maximum update depth exceeded.

### DISCUSSION THREAD

**Discussion by deepakchandrawanshi1063** [react-hook-form/react-hook-form#6895] [Q&A]
Error
Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
![image](https://user-images.githubusercontent.com/41387081/138873395-ddf491c9-f510-4fa2-9717-ef8761592ead.png)

codesandbox link : **https://codesandbox.io/s/maxium-depth-error-16w9h**

getting error while going to edit profile from three-dot(:), while running on local I am getting a prefilled form of PersonalInfo but getting error while clicking form onSubmit  button (Update) that maximum update depth exceeds.

dilaog is opened with form in local but geting error after click on update :-
![image](https://user-images.githubusercontent.com/41387081/138873898-93f272b2-1377-433f-86ea-aca0cec37ccf.png)

error on localhost:-
![image](https://user-images.githubusercontent.com/41387081/138873286-e392a317-548a-4661-9953-6866a6a718e3.png)

my requirement is, after clicking on edit profile. Personalnfo form render in material-ui dialog. after click on form onSubmit .validation should be ferformed and data is send back to backend using axios.

pls help me to solve this issue.

**Comment by lundn** [comment]
A bit hard to follow your code, but I believe your culprit is the `reset(profileData);` in `EditProfile.jsx`. I think you should initialize  `useForm` with `defaultValues` instead of using reset in the form context consuming component.
